runtime/launcher.m

```diff
@@ -497,6 +497,7 @@
 		
 #ifdef DYNAMIC_MONO_RUNTIME
 		config_path = [[NSString stringWithUTF8String: mono_runtime_prefix] stringByAppendingPathComponent: @"/etc/mono/config"];
+		machine_config_path = [[NSString stringWithUTF8String: mono_runtime_prefix] stringByAppendingPathComponent: @"etc/mono/4.5/machine.config"];
```

Comment:
hmm... we'll need to dig the git history behind this. E.g. there are null checks `machine_config_path` later (e.g. https://github.com/xamarin/xamarin-macios/pull/3290/files#diff-af570d5ca3d85213e06ec471ce874414L511) which seems to indicate this behaviour was intended.

The earlier comment (line 491) also seems to skip it - without mentioning why.

Reply:
Yes, I noticed that too, but without a `machine.config` being loaded, System.Configuration just crashes with: https://gist.github.com/Therzok/d0d95a847108f508fb43bd435b5c0249

Reply:
If there is a way to figure out which .net framework to use (2.0, 4.0, 4.5), we can use that instead.

Reply:
Once upon a time the two code paths (statically linked mono vs dynamically loading system mono) were entirely different. Then the code was merged, and ifs added to separate different logic, and that's how the null check for `machine_config_path` came into existence: the code that dynamically loaded the system mono just didn't load the machine.config manually (https://github.com/xamarin/maccore/commit/cedbd33a14480403646ea58847893c2cf6aab40b)

However, I wonder if normal XM apps just don't load a `machine.config` or if they load a different one.

In any case I think this is something that can cause subtle changes in behavior, and somehow needs to be opt-in/opt-out.

Reply:
It seems that mmp copies over this file into the bundle:
https://github.com/xamarin/xamarin-macios/blob/af03020eef61ee4c4dd0b11a3f43abfab14a94bf/tools/mmp/driver.cs#L1770-L1780